src/Stratis.Bitcoin.Features.Api/ApiFeature.cs

```diff
@@ -29,40 +30,44 @@ public sealed class ApiFeature : FullNodeFeature
 
         private IWebHost webHost;
 
+        private readonly ICertificateStore certificateStore;
+
         public ApiFeature(
             IFullNodeBuilder fullNodeBuilder,
             FullNode fullNode,
             ApiFeatureOptions apiFeatureOptions,
             ApiSettings apiSettings,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            ICertificateStore certificateStore)
         {
             this.fullNodeBuilder = fullNodeBuilder;
             this.fullNode = fullNode;
             this.apiFeatureOptions = apiFeatureOptions;
             this.apiSettings = apiSettings;
+            this.certificateStore = certificateStore;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
         }
 
         public override void Initialize()
         {
             this.logger.LogInformation("API starting on URL '{0}'.", this.apiSettings.ApiUri);
-            this.webHost = Program.Initialize(this.fullNodeBuilder.Services, this.fullNode, this.apiSettings);
+            this.webHost = Program.Initialize(this.fullNodeBuilder.Services, this.fullNode, this.apiSettings, this.certificateStore);
+
+            if (this.apiSettings.KeepaliveTimer == null)
+                return;
 
             // Start the keepalive timer, if set.
             // If the timer expires, the node will shut down.
-            if (this.apiSettings.KeepaliveTimer != null)
+            this.apiSettings.KeepaliveTimer.Elapsed += (sender, args) =>
```

Comment:
I don't understand your comment, do you think this changes the behaviour of the Initialize method ? to me this is just a little reduction in nesting.

Reply:
Yes, it is totally unrelated to this PR. It is just making a code look better per your personal preference. So someone else will come later and change it back because their preference will be different from yours. It makes nonsense, it doesn't belong to this PR, why don't you just focus on the issue?